### PR TITLE
Add astyle test to Github Workflow

### PR DESCRIPTION
**Pull request by cothan**
[//]: # (SPDX-License-Identifier: CC-BY-4.0)
[//]: # (TODO Customize PR template)

[//]: # (See https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository )

This PR will enforce future code to be formatted by Astyle, as provision by `.astylerc` at the root of the project. 

* [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)

### CODE REVIEW

**Comment by cothan** [comment]
This PR should fail until the other PR format the whole repo with astyle.

**Comment by mkannwischer** [comment]
A problem we faced for https://github.com/pq-code-package/mlkem-c-embedded is that different versions of astyle format the code differently.
I see two options to solve this:
- We fix to the version that the Github runners have (and document it in the CONTRIBUTING documentation)
- We use something like nix to make sure we have full control over which version is used. 

Both are fine for me for now.

**Comment by cothan** [comment]
The ubuntu-latest provide Astyle 3.1, while in other project my astyle version is 3.4.10. 
So I add Nix package, same as what `mlkem-c-embedded` provide.

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      perSystem = { pkgs, system, inputs', ... }:
+        let
+          core = with pkgs; [
+            # formatter & linters
+            astyle # 3.4.10
+
+            # build dependencies
+            gcc-arm-embedded-13 # arm-gnu-toolchain-13.2.rel1
+            openocd # 0.12.0
+            python311Packages.pyserial # 3.5
```
Comment (mkannwischer) [inline flake.nix:28]: Those are not needed for the AArch64 repo. Please remove.

Comment (cothan) [reply flake.nix]: Done in https://github.com/pq-code-package/mlkem-c-aarch64/pull/24/commits/e55890ea93839417830ae6db2b1c7d8e1dc08716

**Comment by potsrevennil** [comment]
Hi, I came across this PR, I saw that you're using the nix CI setup step from `mlkem-c-embedded` `main` branch. I've did some refactor on the nix setup in CI since the original one is a bit mess, although not yet merged into the main branch, I would recommend to use this refactored version instead.

https://github.com/pq-code-package/mlkem-c-embedded/blob/bde05c9a0b4bbc902a889d82eb6f9913a1181117/.github/actions/setup-nix/action.yaml

**Comment by cothan** [comment]
hi @potsrevennil , nice, thanks for the action.yaml, I will add it immediately.

**Review by mkannwischer — APPROVED** [review]
Thanks. LGTM

**Pull request merged**